orca-echo/src/main/java/com/netflix/spinnaker/orca/echo/util/ManualJudgmentAuthorization.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2020 OpsMx, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.echo.util;
+
+import static java.lang.String.format;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Strings;
+import com.google.common.collect.Sets;
+import com.netflix.spinnaker.fiat.model.UserPermission;
+import com.netflix.spinnaker.fiat.model.resources.Role;
+import com.netflix.spinnaker.fiat.shared.FiatPermissionEvaluator;
+import com.netflix.spinnaker.fiat.shared.FiatStatus;
+import com.netflix.spinnaker.kork.exceptions.SpinnakerException;
+import com.netflix.spinnaker.orca.front50.Front50Service;
+import com.netflix.spinnaker.orca.front50.model.Application;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import retrofit.RetrofitError;
+
+@Component
+public class ManualJudgmentAuthorization {
+  private final Logger log = LoggerFactory.getLogger(getClass());
+
+  private final Front50Service front50Service;
+  private final FiatPermissionEvaluator fiatPermissionEvaluator;
+
+  private final FiatStatus fiatStatus;
+  private final ObjectMapper objectMapper;
+
+  @Autowired
+  public ManualJudgmentAuthorization(
+      Optional<Front50Service> front50Service,
+      Optional<FiatPermissionEvaluator> fiatPermissionEvaluator,
+      FiatStatus fiatStatus,
+      ObjectMapper objectMapper) {
+    this.front50Service = front50Service.orElse(null);
+    this.fiatPermissionEvaluator = fiatPermissionEvaluator.orElse(null);
+
+    this.fiatStatus = fiatStatus;
+    this.objectMapper = objectMapper;
+  }
+
+  /**
+   * A manual judgment will be considered "authorized" if the current user has at least one of the
+   * required judgment roles _and_ read/write/execute permissions on the application being judged.
```

Comment:
I still question why manual judgment authorization should care about application permissions. 

This comment has been left as-is to reflect the intent of the original PR but I believe it should be revisited.